pkg/build/builder/docker.go

```diff
@@ -83,21 +84,35 @@ func (d *DockerBuilder) Build() error {
 		push = true
 	}
 
-	if err := d.dockerBuild(buildDir, d.build.Spec.Source.Secrets); err != nil {
+	buildTag := randomBuildTag(d.build.Namespace, d.build.Name)
+
+	if err := d.dockerBuild(buildDir, buildTag, d.build.Spec.Source.Secrets); err != nil {
+		return err
+	}
+
+	cname := containerName("docker", d.build.Name, d.build.Namespace, "post-commit")
+	if err := execPostCommitHook(d.dockerClient, d.build.Spec.PostCommit, buildTag, cname); err != nil {
```

Comment:
seems like we'd still want to removeImage(buildTag) if we get an error here, right?  or are you thinking it's worth leaving it around for debug purposes?